### PR TITLE
Minimal value of minimal_quantity set to 1

### LINKED ISSUE #18850: Minimal quantity of 0 on product cause add to cart 0
#### Describe the bug

When the minimal quantity of a product is set to 0 (disable the fonctionality) in the B.O. the default value in add to cart part is by default 0… 

#### Expected behavior

The quantity to add to cart must be at least 1. There is no reason a customer order 0 product when it clicks on the «Add to cart» button …

#### Steps to Reproduce

1. Go to B.O. > Catalog > Product
2. Click on 'Quantities'
3. Set minimal quantity to 0
4. Go to F.O. > and show the Product detail page
5. The add to cart input has a value of «0» 

**Screenshots**

image
image

#### Additional information

* PrestaShop version: see on 1.7.5.0 / 1.7.6.2
* PHP version: 5.6.30 ( for the 1.7.6.2) / 7.0.33 (for the 1.7.5.0)

### PR DESCRIPTION
**Pull request by thalassa-web**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | When a minimal_quantity on a product is set to 0, the add to cart quantity will be 1 (and not 0)
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #18850 
| How to test?  | Set a minimal quantity of 0 on a product and display the product detail in Front Office.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18853)

### CODE REVIEW

**Comment by Progi1984** [comment]
An answer from @PrestaShop/prestashop-product-team is needed in #18850.

File: webservice/dispatcher.php [hunk 1]
```diff
@@ -34,6 +34,11 @@
 Context::getContext()->cart = new Cart();
 Context::getContext()->container = ContainerBuilder::getContainer('webservice', _PS_MODE_DEV_);
 
+// If the currency is not defined in the context, initialize with the default one !
```
Comment (matks) [inline webservice/dispatcher.php:37]: Hi @thalassa-web I believe this line is actually part of another PR, right ? It's not linked to this topic 😉

File: controllers/front/ProductController.php [hunk 2]
```diff
@@ -1089,7 +1089,7 @@ protected function getProductMinimalQuantity($product)
             $minimal_quantity = $this->product->minimal_quantity;
         }
 
-        return $minimal_quantity;
+        return max($minimal_quantity, 1);
```
Comment (matks) [inline controllers/front/ProductController.php:1092]: Warning: this function is used in multiple places. Modifying its behavior can create side-effects (for example with the computations related to check whether user can order X amount on product A provided it has stock B).

So we better have extensive checks 🤔

File: webservice/dispatcher.php (hunk 1 above)
Comment (thalassa-web) [reply webservice/dispatcher.php:37]: Yes you're right … Sorry …

**Comment by PierreRambaud** [comment]
Hi @thalassa-web ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**